homeassistant/components/vicare/number.py

```diff
@@ -146,6 +153,20 @@ def update(self) -> None:
                 self._attr_native_value = self.entity_description.value_getter(
                     self._api
                 )
+                if (min_fn := self.entity_description.min_value_getter) and (
+                    min_value := min_fn(self._api)
+                ):
+                    self._attr_native_min_value = min_value
+
+                if (max_fn := self.entity_description.max_value_getter) and (
+                    max_value := max_fn(self._api)
+                ):
+                    self._attr_native_max_value = max_value
+
+                if (stepping_fn := self.entity_description.stepping_getter) and (
+                    stepping_value := stepping_fn(self._api)
+                ):
+                    self._attr_native_step = stepping_value
```

Comment:
Please create a helper function:
```python
def get_value(fn: Callable[[PyViCareDevice], float | None], api: PyViCareHeatingDeviceComponent): -> float | None:
    if fn:
        return fn(api)

    return None
```

and we call it here three times:
```suggestion
                self._attr_native_min_value = get_value(self.entity_description.min_value_getter, self._api)
                self._attr_native_max_value = get_value(self.entity_description.max_value_getter, self._api)
                self._attr_native_step = get_value(self.entity_description.stepping_getter, self._api)
```

Sorry did not see it yesterday

Reply:
Yes, this looks better 👍

Reply:
Cannot assign `float | None` to `_attr_native_max_value` etc.